Taskfile.yml

```
@@ -0,0 +1,87 @@
+version: "3"
+
+includes:
+  utils: "tools/yscope-dev-utils/taskfiles/utils.yml"
+
+vars:
+  # Paths
+  G_BUILD_DIR: "{{.ROOT_DIR}}/build"
+  G_EMSCIPRTEN_DIR: "{{.G_BUILD_DIR}}/emsdk"
+  G_WSAM_BUILD_DIR: "{{.G_BUILD_DIR}}/wasm"
+
+tasks:
+  default:
+    deps: ["wasm"]
+
+  clean:
+    cmds:
+      - task: "clean-emscripten"
+      - task: "clean-wasm"
+      - "rm -rf '{{.G_BUILD_DIR}}'"
+
+  clean-emscripten:
+    cmds:
+      - "rm -rf '{{.G_EMSCIPRTEN_DIR}}'"
+
+  clean-wasm:
+    cmds:
+      - "rm -rf '{{.G_WSAM_BUILD_DIR}}'"
+
+  emscripten:
+    vars:
+      CHECKSUM_FILE: "{{.G_BUILD_DIR}}/{{.TASK}}.md5"
+      OUTPUT_DIR: "{{.G_EMSCIPRTEN_DIR}}"
+    sources: ["{{.TASKFILE}}"]
+    generates: ["{{.CHECKSUM_FILE}}"]
+    deps:
+      - "init"
+      - task: "utils:validate-checksum"
+        vars:
+          CHECKSUM_FILE: "{{.CHECKSUM_FILE}}"
+          DATA_DIR: "{{.OUTPUT_DIR}}"
+    cmds:
+      - task: "clean-emscripten"
+      - "git clone 'https://github.com/emscripten-core/emsdk.git' '{{.G_EMSCIPRTEN_DIR}}'"
+      - |-
+        cd "{{.G_EMSCIPRTEN_DIR}}"
+        ./emsdk install latest
+        ./emsdk activate latest
+      # This command must be last
+      - task: "utils:compute-checksum"
+        vars:
+          DATA_DIR: "{{.OUTPUT_DIR}}"
+          OUTPUT_FILE: "{{.CHECKSUM_FILE}}"
+
+  wasm:
+    vars:
+      CHECKSUM_FILE: "{{.G_BUILD_DIR}}/{{.TASK}}.md5"
+      OUTPUT_DIR: "{{.G_WSAM_BUILD_DIR}}"
+    sources:
+      - "{{.TASKFILE}}"
+      - "CMakeLists.txt"
+      - "src/**/*"
+    generates: ["{{.CHECKSUM_FILE}}"]
+    deps:
+      - "emscripten"
+      - task: "utils:validate-checksum"
+        vars:
+          CHECKSUM_FILE: "{{.CHECKSUM_FILE}}"
+          DATA_DIR: "{{.OUTPUT_DIR}}"
+    cmds:
+      - task: "clean-wasm"
+      - "mkdir '{{.OUTPUT_DIR}}'"
+      - |-
+        cmake -S "{{.ROOT_DIR}}" -B "{{.OUTPUT_DIR}}" -G "Unix Makefiles"
+        cmake --build "{{.OUTPUT_DIR}}" --parallel
+      # This command must be last
+      - task: "utils:compute-checksum"
+        vars:
+          DATA_DIR: "{{.OUTPUT_DIR}}"
+          OUTPUT_FILE: "{{.CHECKSUM_FILE}}"
+
+  init:
+    internal: true
+    silent: true
+    run: "once"
+    cmds:
+      - "mkdir -p '{{.G_BUILD_DIR}}'"
```

```suggestion
    cmds: ["mkdir -p '{{.G_BUILD_DIR}}'"]
```

Just to save some lines.